# Проверка статистических гипотез. Доверительные интервалы. A/B-тестирование

__1. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.__

Из условий задачи: $H_0: a = 17 мм,\quad\alpha = 0.025,\quad\Omega_\alpha = (t_{1 - \alpha}, \infty),\quad n=100,\quad a=17.5,\quad\sigma^2 = 4 \Rightarrow\sigma = 2$

Для проверки гипотезы используем следующую статистику: $T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}},$ т.к. дисперсия нам известна. Примем, что распределние нормальное и $H_1: a \ne 17 мм$

In [1]:
import numpy as np
from scipy import stats

In [2]:
a = 17
alpha = 0.025
n = 100
a_mean = 17.5
sigma = 2

T = (a_mean - a)/(sigma/np.sqrt(n))
print(T)

2.5


Определим правостороннюю критическую область: $\Omega_\alpha = (t_{1 - \alpha}, \infty)$

In [3]:
t = stats.norm.ppf(1 - alpha)
print(t, '=~', round(t, 3))

1.959963984540054 =~ 1.96


Т.к. значение статистики $T=2.5$ попадает в критическую область $\Omega_\alpha = (1.96, \infty)$, то верна гипотеза $H_1$

__2. Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.__

Из условий задачи: $\sigma = 16,\quad p=0.95,\quad\overline{X}=80,\quad n=256$

Построим доверительный интервал с помощью квантилей стандартного распределения, т.к. дисперсия нам известна $(\sigma^2)$
$
P \left(
    \overline{X} - t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq a \leq 
    \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$

In [4]:
sigma = 16
p = 0.95
alpha = 1 - p
a_mean = 80
n = 256

t = stats.norm.ppf(1 - alpha / 2)
print(t)

1.959963984540054


In [5]:
print(f'Мат.ожидание в интервале {a_mean - t * sigma / np.sqrt(n), a_mean + t * sigma / np.sqrt(n)} с вероятностью {p}')

Мат.ожидание в интервале (78.04003601545995, 81.95996398454005) с вероятностью 0.95


__3. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:__

__202, 203, 199, 197, 195, 201, 200, 204, 194, 190__

__Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?__

Из условий задачи: $H_0: a = 200 г,\quad n=10,\quad\alpha = 0.01$

Для проверки гипотезы используем следующую статистику: $T(X) = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}},$ т.к. дисперсия нам неизвестна, следовательно критическая область $\Omega_\alpha =\left( -\infty, - t_{1 -\alpha/2,\: n - 1}\right)\cup\left( t_{1 -\alpha/2,\: n - 1},\infty\right)$. И примем $H_1: a \ne 200 г$

In [6]:
samples = (202, 203, 199, 197, 195, 201, 200, 204, 194, 190)
a = 200
alpha = 0.01
n = 10

X_mean = np.mean(samples)
sigma_unb = np.std(samples, ddof=1) #несмещённая оценка среднего квадратического отклонения
T = (X_mean - a)/(sigma_unb/np.sqrt(n))
print(T)

-1.0651074037450896


Определим критическую область

In [7]:
t = stats.t.ppf(1 - alpha / 2, df=n - 1)
print(t, "=~", round(t, 3))

3.2498355440153697 =~ 3.25


Т.к. гипотеза $H_0$ не входит в критическую область $\Omega_\alpha =\left( -\infty, - 3.25\right)\cup\left(3.25,\infty\right)$,
то она скорее всего верна.